compiler/resolution/functionResolution.cpp

```diff
@@ -10514,15 +10536,20 @@ static void resolvePrimInit(CallExpr* call, Symbol* val, Type* type) {
 
   if (call->id == breakOnResolveID) gdbShouldBreakHere();
 
-  AggregateType* at     = toAggregateType(type);
-
+  AggregateType* at = toAggregateType(type);
   val->type = type;
 
   // Shouldn't be called with ref types
   INT_ASSERT(at == NULL || at->getValType() == at);
 
   SET_LINENO(call);
 
+  if(type == dtVoid && !fatalErrorsEncountered())
```

Comment:
When `--ignore-errors-for-pass` is used, earlier errors, such as `compilerError` errors, are marked as pending until the end of the resolution pass, which can cause secondary errors about `void` variables here. If the function resolution pass has any pending fatal errors, we suppress any errors about `void` variables here, which might be secondary to the original fatal error.